services/checks/checks.go

```diff
@@ -474,49 +477,82 @@ func (s *Service) minPMMAgentVersion(t check.Type) *version.Parsed {
 	}
 }
 
-// executeChecks runs all available checks for all reachable services. Interval specifies from what
-// group to execute checks, empty interval means "execute everything".
-func (s *Service) executeChecks(ctx context.Context, interval check.Interval) error {
-	s.l.Infof("Executing %s checks...", interval)
+// filterChecks filters checks by several parameters. If group specified then only matched checks will be returned,
+// empty group means `any interval`. If enable slice is specified then only matched checks will be returned, empty
+// enable slice means `all enabled`. Checks specified in disabled slice are skipped, empty `disabled` slice means
+// `nothing disabled`.
+func (s *Service) filterChecks(checks []check.Check, group check.Interval, disable, enable []string) []check.Check {
+	var res []check.Check
+	disableMap := make(map[string]struct{}, len(disable))
+	for _, e := range disable {
+		disableMap[e] = struct{}{}
+	}
+
+	enableMap := make(map[string]struct{}, len(enable))
+	for _, e := range enable {
+		enableMap[e] = struct{}{}
+	}
+
+	for _, c := range checks {
+		// If empty group passed, which means `any group`
+		// or check has required interval
+		// or check has empty interval and required interval is `standard`.
+		if group == "" || c.Interval == group || (group == check.Standard && c.Interval == "") {
```

Comment:
Maybe also negate this condition and add continue.
"Align the happy path to the left edge"

Reply:
Yes, I know this function in not so easy to read. I tried different approaches, even split it to different functions. If we will nagate conditions it will be harder to understand the logic, so nested `if` is a lesser evil.